server/api/attendance/attendance.controller.js

```diff
@@ -251,7 +251,55 @@ var getAttendance = function(userId, classYearId, cb){
   var callback = cb || function(){};
   Attendance.find({user:userId, classYear:classYearId}, function (err, attendance) {
     if (err) {return handleError(err)}
-    return callback(attendance);
+    SmallGroup.find({classYear:classYearId, students:userId}, function (err, smallgroup) {
+      if (err) {return handleError(err)}
+      if (typeof(smallgroup[0]) != 'undefined') {
+        for (var i = 0; i < smallgroup[0]["dayCodes"].length; i++) {
+          var smallattend = smallgroup[0]["dayCodes"][i];
+          var smalldate = new Date(smallattend.date);
+          smalldate.setUTCHours(4); //sets time to midnight of EDT zone
+          smalldate.setUTCMinutes(0);
+          smalldate.setUTCSeconds(0);
+          smalldate.setUTCMilliseconds(0);
+
+          var found = false;
+          for (var j = 0; j < attendance.length; j++) {
+            var attend = attendance[j];
+            if (attend.date.getTime() == smalldate.getTime()) {
+              if (attend.bonusDay === smallattend.bonusDay) {
+                if (attend.smallgroup === true) {
+                  found = true;
+                  break;
+                }
+              }
+            }
+          }
+          if (!found) {
+            attendance.push({date:smallattend.date, bonusDay:smallattend.bonusDay, smallgroup:true, verified:false, present:false});
+          }
+        }
+      }
+    });
+    ClassYear.find({current:true}, function (err, classyear) {
+      if (err) {return handleError(err)}
+      if (typeof(classyear) != 'undefined') {
+        for (var i = 0; i < classyear[0]["dayCodes"].length; i++) {
```

Comment:
Instead of classyear[0] you should be able to use Mongooses `findOne` and a check for null instead of the check of typeof too